apollo-api/src/main/java/com/spotify/apollo/Response.java

```diff
@@ -40,10 +42,33 @@
 
   /**
    * The response headers.
+   * Deprecated in favor of {@link Response#headerEntries()}
    */
+  @Deprecated
   Map<String, String> headers();
 
   /**
+   * The response headers
+   */
+  Iterable<Map.Entry<String, String>> headerEntries();
```

Comment:
Why Iterable here? Since we're spending some effort on retaining order of headers, would it not make more sense to return a List?